lib/apiservers/engine/backends/image_cache.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vicbackends
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+
+	derr "github.com/docker/docker/errors"
+	"github.com/docker/engine-api/types"
+	"github.com/vmware/vic/lib/apiservers/portlayer/client/storage"
+	"github.com/vmware/vic/lib/apiservers/portlayer/models"
+	"github.com/vmware/vic/lib/guest"
+	"github.com/vmware/vic/lib/metadata"
+)
+
+// ImageCache is an in-memory cache of image layers. It is refreshed at startup
+// by a call to the portlayer. It is updated when new layers are pulled or
+// layers are deleted.
+type ImageCache struct {
+	sync.Mutex
```

Comment:
Alternatively, if you think read is the hot path, you can convert this to RWMutex and grab the write lock in Update and read lock in Get